benefits/core/templates/core/includes/modal-trigger-link.html

```diff
@@ -0,0 +1,2 @@
+<!-- Inline link to trigger modal -->
+<a href="#{{ id }}" class="{{ classes }}" data-bs-toggle="modal" data-bs-target="#{{ id }}">{{ text }}</a>
```

Comment:
There is currently a mobile design bug on the modal trigger on dev. I don't think this modal trigger link was mobile tested thoroughly - this is what it looks like now on dev:

<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/5a50ea10-6175-4e6b-9c9a-c6af2d19ec57">

And it looks even sillier with 2 links like this. This is why I decided to switch from `button` to `a`.

A `button` - even with all sorts of CSS styles applied - cannot be styled to break lines naturally like a link can. 

The modal trigger link as an `a` works fine design-wise/mobile-wise for short links (like LittlePay and Login.gov): 
<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/76c321b9-70c8-41c4-a679-a39fd1828eef">
<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/134cca14-eec9-4502-b95f-01332777a90b">

But the modal trigger link does NOT style responsively correctly on mobile for the long text links on the Eligibility Index page, when using the `button` element.

<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/5a50ea10-6175-4e6b-9c9a-c6af2d19ec57">
<img width="1512" alt="image" src="https://github.com/cal-itp/benefits/assets/3673236/06acd94e-dc98-4392-bef8-621e3079ac9a">

Originally I tried using a new-ish CSS style called `display: contents` which kinda works but it entirely removes the link from the a11y panel, won't go into focus, won't take an underline, etc., so it's not an adequate solution. 

USDWS Modal Guidance (https://designsystem.digital.gov/components/modal/#guidance) says we can use links as openers. The main keyboard requirement for links as openers is that it opens on enter (but not spacebar). Which this code complies with. And on second thought, since it visually _looks_ like a text link - I realized it is better as a text link semantically. Given the design of a link, I'm coding it as a link.

(Although I think if I were designing this, I would *design* it as a button and use button code semantics, personally speaking. For this and the Go back button too. https://a11y-solutions.stevenwoodson.com/solutions/navigation/link-anti-patterns/ https://www.a11yproject.com/posts/creating-valid-and-accessible-links/)

Reply:
Additionally, @angela-tran pointed out that using an `a` here will allow us to track link open stats on Amplitude.